osh/client/commands/cmd_mock_build.py

```diff
@@ -15,6 +16,16 @@ def options(self):
 
         # this option cannot be used for diff-build tasks
         add_tarball_option(self.parser)
+        add_dist_git_url_option(self.parser)
+
+    def prepare_task_options(self, args, kwargs):
+        # Check if exactly one of SRPM, NVR, or dist-git URL is provided
+        if sum([bool(args), bool(kwargs.get("nvr")), bool(kwargs.get("git_url"))]) != 1:
+            self.parser.error("please specify either SRPM or NVR or dist-git URL")
+
+        self._process_srpm_option(args, kwargs)
```

Comment:
Can we add it to the `def options`, so that it looks more uniform

Reply:
It requires `args` and `kwargs` to be present for some validation, it may also populate the `kwargs`.

`options()` has no `args`/`kwarg` available, moving it there is probably not a good idea unless some careful refinement.